src/System.Drawing.Primitives/tests/ColorTests.cs

```diff
@@ -491,5 +492,45 @@ public void DebuggerAttributesAreValid()
             DebuggerAttributes.ValidateDebuggerDisplayReferences(Color.Aquamarine);
             DebuggerAttributes.ValidateDebuggerDisplayReferences(Color.FromArgb(4, 3, 2, 1));
         }
+
+        [Fact]
+        public void UserPreferenceChangingEventTest()
+        {
+            int element = 12; // Win32SystemColors.AppWorkSpace.
+            Color oldColor = System.Drawing.SystemColors.AppWorkspace;
+
+            // A call to ToArgb is necessary before changing the system colors because it initializes the knownColorTable.
+            int oldColorArgb = oldColor.ToArgb();
+            int oldColorAbgr = GetColorRefValue(oldColor);
+
+            Color newColor = oldColor != Color.Gold ? Color.Gold : Color.Silver;
+            int newColorArgb = newColor.ToArgb();
+            int newColorAbgr = GetColorRefValue(newColor);
+
+            Assert.NotEqual(newColorArgb, oldColorArgb);
+
+            try
+            {
+                Assert.Equal(1, SetSysColors(1, new int[] { element }, new int[] { newColorAbgr }));
+
+                RetryHelper.Execute(() =>
+                {
+                    Assert.Equal(newColorArgb, oldColor.ToArgb());
+                });
+            }
+            finally
+            {
+                Assert.Equal(1, SetSysColors(1, new int[] { element }, new int[] { oldColorAbgr }));
+            }
+        }
+
+        [DllImport("user32.dll", SetLastError = true)]
```

Comment:
You will need to include this test only on Windows as `user32.dll` will not be available on Unix. 

@ericstj any other idea?

Reply:
Also likely not working on Nano. Go have a look at the attributes on corresponding SystemEvents tests to see what to do.